src/libgeodecomp/loadbalancer/loadbalancer.h

```diff
@@ -4,14 +4,20 @@
 #include <libgeodecomp/misc/stdcontaineroverloads.h>
 #include <vector>
 
+#if defined(LIBGEODECOMP_WITH_HPX)
+#include<hpx/config.hpp>
+#elif !defined(HPX_COMPONENT_EXPORT)
+#define HPX_COMPONENT_EXPORT
```

Comment:
Or, since a lot of classes are getting this treatment, rather have a macro LGD_HPX_COMPONENT_MACRO which then can either expand to an empty string or to HPX_COMPONENT_MACRO